spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkV2Filters.java

```diff
@@ -302,10 +354,25 @@ private static <T> T childAtIndex(Predicate predicate, int index) {
     return (T) predicate.children()[index];
   }
 
+  private static boolean couldConvert(org.apache.spark.sql.connector.expressions.Expression expr) {
+    return isRef(expr) || isSystemFunc(expr);
+  }
+
   private static boolean isRef(org.apache.spark.sql.connector.expressions.Expression expr) {
     return expr instanceof NamedReference;
   }
 
+  private static boolean isSystemFunc(org.apache.spark.sql.connector.expressions.Expression expr) {
+    if (expr instanceof UserDefinedScalarFunc) {
+      UserDefinedScalarFunc udf = (UserDefinedScalarFunc) expr;
+      return udf.canonicalName().startsWith("iceberg")
```

Comment:
Do we need this string check? I feel like all of the functions should be listed in the list on 369?

Reply:
There could be possible some other UDF functions like 'catalog.other.years'.

Reply:
I think the problem is that this is using the UDF's `canonicalName` but the other check uses the function's `name`. Those can differ: `BucketFunction.name()` return `bucket`, what the user would call, but the canonical function name identifies the exact bound function no matter how it is loaded so `BucketInt.canonicalName()` returns `iceberg.bucket(int)`.

If we want to limit to just the Iceberg-defined functions, then this is necessary. It may be better to have a set of supported functions that uses just the `canonicalName`.

Reply:
While that's not a constant, there are a limited number of values that `sqlType.catalogString()` might return.

Reply:
I have tried to do that, however `TruncateDecimal` needs precision and scale which are runtime values.
```java
    @Override
    public String canonicalName() {
      return String.format("iceberg.truncate(decimal(%d,%d))", precision, scale);
    }
```

Reply:
You can similarly enumerate them, but this isn't a big deal.